homeassistant/components/bang_olufsen/config_flow.py

```diff
@@ -0,0 +1,162 @@
+"""Config flow for the Bang & Olufsen integration."""
+from __future__ import annotations
+
+import ipaddress
+import logging
+from typing import Any, TypedDict
+
+from aiohttp.client_exceptions import ClientConnectorError
+from mozart_api.exceptions import ApiException
+from mozart_api.mozart_client import MozartClient
+import voluptuous as vol
+
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_MODEL
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import SelectSelector, SelectSelectorConfig
+
+from .const import (
+    ATTR_FRIENDLY_NAME,
+    ATTR_ITEM_NUMBER,
+    ATTR_SERIAL_NUMBER,
+    ATTR_TYPE_NUMBER,
+    COMPATIBLE_MODELS,
+    CONF_SERIAL_NUMBER,
+    DEFAULT_MODEL,
+    DOMAIN,
+)
+
+
+class EntryData(TypedDict, total=False):
+    """TypedDict for config_entry data."""
+
+    host: str
+    jid: str
+    model: str
+    name: str
+
+
+class BangOlufsenConfigFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    _beolink_jid = ""
+    _client: MozartClient
+    _host = ""
+    _model = ""
+    _name = ""
+    _serial_number = ""
+
+    def __init__(self) -> None:
+        """Init the config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is not None:
+            self._host = user_input[CONF_HOST]
+            self._model = user_input[CONF_MODEL]
+
+            # Check if the IP address is a valid address.
+            try:
+                ipaddress.ip_address(self._host)
```

Comment:
We normally allow either a hostname or an IP-address, is there some reason we only allow IP-addresses here?